### ISSUE TITLE
UnusedPrivateField false positive on typed private fields

### ISSUE THREAD

**Issue by timosmit** [phpmd/phpmd#729]
- PHPMD version: 2.8.1
- PHP Version: 7.4.2
- Installation type: composer
- Operating System / Distribution & Version: Arch Linux (rolling release)

### Current Behavior
PHPMD detects a typed private field in an entity class as unused (UnusedPrivateField rule).

### Expected Behavior
PHPMD should not detect the field as unused, since it is used by several getters and setters.

### Steps To Reproduce:
I updated my Symfony project (version 5.0.4) and PHP executable (version 7.4.2) and decided to add typing to the properties in my entities. The following code should be enough to reproduce the issue:

```
/**
 * @ORM\Entity(repositoryClass="App\Repository\MileageRepository")
 */
class Mileage
{
    private int $id;

    /**
     * @ORM\Id()
     * @ORM\GeneratedValue()
     * @ORM\Column(type="integer")
     */
    public function getId(): ?int
    {
        return $this->id;
    }
}
```

Note that the class above makes use of Doctrine's entity annotations and the id field should be filled by the Doctrine entity manager.

**Comment by ossinkine** [comment]
Should be fixed in https://github.com/pdepend/pdepend/pull/453, try to update `pdepend/pdepend` up to 2.7.1

**Comment by kylekatarnls** [comment]
Sadly no, it still happen because it's not related to typed property (note that it also happen with untyped properties).

This is more likely related to a more global issue about detection read and write of a variable/member: https://github.com/pdepend/pdepend/issues/444

**Comment by kylekatarnls** [comment]
@ossinkine You were absolutely right, it has been fixed by PDepend 2.7.1. I misunderstood the problem. I will add a test to ensure it remains fixed.

**Comment by kylekatarnls** [comment]
However typed property are not properly detected when private and unused (false negative), but this will be fixed with https://github.com/pdepend/pdepend/issues/474
